test/python/transpiler/test_mappers.py

```diff
@@ -0,0 +1,280 @@
+         q0:----------m-----|--
+                      |     |
+         q1:-[H]-(+)--|-m---|--
+                  |   | |   |
+         q2:------.---|-|-m-|--
+                      | | | |
+         c0:----------.-|-|-|--
+         c1:------------.-|-|--
+         c2:--------------.-|--
+         c3:----------------.--
+         CouplingMap map: [1]<-[0]->[2]->[3]
+
+        expected count: '000': 50%
+                        '110': 50%
+        """
+        self.counts = {'0000': 512, '0110': 512}
+        self.shots = 1024
+        self.delta = 5
+        coupling_map = [[0, 1], [0, 2], [2, 3]]
+
+        qr = QuantumRegister(4, 'q')
+        cr = ClassicalRegister(4, 'c')
+        circuit = QuantumCircuit(qr, cr, name='initial_layout')
+        circuit.h(qr[1])
+        circuit.cx(qr[1], qr[2])
+        circuit.measure(qr, cr)
+
+        layout = [qr[3], qr[0], qr[1], qr[2]]
+
+        result = transpile(circuit, self.create_backend(), coupling_map=coupling_map,
+                           pass_manager=self.create_passmanager(coupling_map, layout))
+        self.assertResult(result, circuit.name)
+
+    def test_handle_measurement(self):
+        """Handle measurement correctly.
+
+         q0:--.-----(+)-m-------
+              |      |  |
+         q1:-(+)-(+)-|--|-m-----
+                  |  |  | |
+         q2:------|--|--|-|-m---
+                  |  |  | | |
+         q3:-[H]--.--.--|-|-|-m-
+                        | | | |
+         c0:------------.-|-|-|-
+         c1:--------------.-|-|-
+         c2:----------------.-|-
+         c3:------------------.-
+
+         CouplingMap map: [0]->[1]->[2]->[3]
+
+        expected count: '0000': 50%
+                        '1011': 50%
+        """
+        self.counts = {'1011': 512, '0000': 512}
+        self.shots = 1024
+        self.delta = 5
+        coupling_map = [[0, 1], [1, 2], [2, 3]]
+
+        qr = QuantumRegister(4, 'q')
+        cr = ClassicalRegister(4, 'c')
+        circuit = QuantumCircuit(qr, cr, name='handle_measurement')
+        circuit.h(qr[3])
+        circuit.cx(qr[0], qr[1])
+        circuit.cx(qr[3], qr[1])
+        circuit.cx(qr[3], qr[0])
+        circuit.measure(qr, cr)
+
+        result = transpile(circuit, self.create_backend(), coupling_map=coupling_map,
+                           pass_manager=self.create_passmanager(coupling_map))
+        self.assertResult(result, circuit.name)
+
+
+class TestsBasicSwap(CommonTestCases, QiskitTestCase):
+    """Test CommonTestCases using BasicSwap."""
+    pass_class = BasicSwap
+
+
+class TestsLookaheadSwap(CommonTestCases, QiskitTestCase):
+    """Test CommonTestCases using LookaheadSwap."""
+    pass_class = LookaheadSwap
+
+
+class TestsStochasticSwap(CommonTestCases, QiskitTestCase):
+    """Test CommonTestCases using StochasticSwap."""
+    pass_class = StochasticSwap
+    additional_args = {'seed': 0}
+
+
+if __name__ == '__main__':
+    CommonUtilitiesMixin.regenerate_expected = True
```

Comment:
In our codebase, running a test as a module is semantically the same as running those test from the test runner.  It’d be OK if we can trigger the regeneration process explicitly. By doing something like:

```bash
$ python -m test.python.transpiler.test_mappers regenerate
```

Reply:
Unittest does not allow parameter:
```
$ python -m test.python.transpiler.test_mappers regenerate
E
======================================================================
ERROR: regenerate (unittest.loader._FailedTest)
----------------------------------------------------------------------
AttributeError: module '__main__' has no attribute 'regenerate'

----------------------------------------------------------------------
Ran 1 test in 0.000s
```

I can argue that we dont have that semantics. Many of the test scripts dont have main code.

Reply:
I can argue that we dont have that semantics. Many of the test scripts dont have main code.

Any way, next snippet added in a69acdde:
```
    if len(sys.argv) >=2 and sys.argv[1] == 'regenerate':
        CommonUtilitiesMixin.regenerate_expected = True
        del sys.argv[1]
```